libqtile/widget/tasklist.py

```diff
@@ -263,7 +263,14 @@ def get_taskname(self, window):
 
     @property
     def windows(self):
-        return self.bar.screen.group.windows
+        if self.qtile.core.name == "wayland":
+            return self.bar.screen.group.windows
+        else:
+            return [
+                w
+                for w in self.bar.screen.group.windows
+                if w.window.get_wm_type() in ("normal", None)
+            ]
```

Comment:
Maybe reverse the if statement because X11 is the special case here (if/when a new backend comes along, the if statement would use the same code path here as wayland):
```suggestion
        if self.qtile.core.name == "x11":
            return [
                w
                for w in self.bar.screen.group.windows
                if w.window.get_wm_type() in ("normal", None)
            ]
	return self.bar.screen.group.windows
```